src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpSyntaxGeneratorInternal.cs

```diff
@@ -128,6 +128,7 @@ internal static SyntaxTokenList GetParameterModifiers(RefKind refKind, bool forF
                 // The return parameter must use ref readonly, like regular methods.
                 RefKind.In when !forFunctionPointerReturnParameter => SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.InKeyword)),
                 RefKind.RefReadOnly when forFunctionPointerReturnParameter => SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.RefKeyword), SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword)),
+                RefKind.RefReadOnlyParameter => SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.RefKeyword), SyntaxFactory.Token(SyntaxKind.ReadOnlyKeyword)),
```

Comment:
Could you also add a test for the following:
- formatting (IDE, probably can be tested in `FormattingTests`) and SyntaxNormalizer (that's in the compiler)
- QuickInfo (IDE, probably can be tested in `SyntacticQuickInfoSourceTests`)

Reply:
note: these can go in other PRs.  We don't expect a single compiler PR that adds all tests.

Reply:
> It may also be interesting to test F1/help. I'm not sure if we're able to go to a `ref readonly` page, or if we could only go to a "ref" or "readonly" page. @akhera99 would you know?

Seems like it should be possible to go to a 'ref readonly' page. https://github.com/dotnet/roslyn/blob/993fcc8ce888ecd3d946001012915b3a5fed757a/src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs#L408

Reply:
Seems that `ref`/`in` modifiers don't have quick info.

Also there's no `ref readonly` page probably (`ref readonly` returns also go separately to `ref` and `readonly`).

Reply:
> Also there's no ref readonly page probably

i have no issue with this.

Reply:
Another interesting scenario to test is automatic interface and override generation. For merging the feature to `main`, we only need it to not crash, but before shipping we'll want it to produce the right modifiers.

Reply:
> Another interesting scenario to test is automatic interface and override generation

That's already tested in this PR, I believe. See ImplementAbstractClassTests and ImplementInterfaceTests